tornado/log.py

```diff
@@ -54,18 +54,20 @@
 
 
 def _stderr_supports_color():
-    color = False
-    if hasattr(sys.stderr, 'isatty') and sys.stderr.isatty():
-        if curses:
-            try:
+    try:
+        if hasattr(sys.stderr, 'isatty') and sys.stderr.isatty():
+            if curses:
                 curses.setupterm()
                 if curses.tigetnum("colors") > 0:
-                    color = True
-            except Exception:
-                pass
-        elif colorama:
-            color = True
-    return color
+                    return True
+            elif colorama:
+                if sys.stderr is getattr(colorama, 'wrapped_stderr', object()):
```

Comment:
Neither appear to be documented, which is what makes me think that using `isinstance` (and failing gracefully when there is an `AttributeError` there) is somewhat better. It also makes it a bit easier to limit it only to Windows, which is the only time we would need to resort to colorama (or otherwise reimplement exactly what colorama already does).

Yes, what you have was in my previous PR. I honestly don't know how that got by. This is one reason I explicitly included the demo in #2014 so that it's at least possible to interactively test.